frontend/src/App.jsx

```diff
@@ -1,37 +1,64 @@
 import './App.css';
 import { useState, useEffect } from 'react';
 import { UserContext } from './UserContext';
+import { LoadingProvider } from './Components/Loading/LoadingContext';
 import { BrowserRouter, Routes, Route } from 'react-router-dom';
 import HomePage from './Components/HomePage/HomePage';
 import LoginForm from './Components/LoginForm/LoginForm';
 import SignupForm from './Components/SignupForm/SignupForm';
+import AddTrip from './Components/HomePage/AddTrip/AddTripPage';
+import Sidebar from './Components/HomePage/Sidebar';
+import PastTrips from './Components/HomePage/PastTrips/PastTrips';
+import { API_BASE_URL } from './config.js';
 
-function App() {
 
-  //storing user data in local storage for further usage in authentication 
-  const [user, setUser] = useState(() => {
-    const storedUser = localStorage.getItem('user');
-    return storedUser ? JSON.parse(storedUser) : null;
-  });
+function App() {
+  const [user, setUser] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
 
   const updateUser = (newUser) => {
     setUser(newUser);
   };
 
   useEffect(() => {
-    localStorage.setItem('user', JSON.stringify(user));
-  }, [user]);
+    //making a call to backend middleware API to check the authentication status of user
+    const checkAuth = async () => {
```

Comment:
Actually, is it possible to perform this authentication in the express router directly? I might have missed that, you can do that directly in express before rendering routes that are supposed to be protected.

Reply:
I'll look into it and let you know any updates